Update assets page layout

ayumitk opened the pull request:
**Pull Request Summary**

ref: https://github.com/AstarNetwork/astar-apps/pull/922

**Check list**

- [ ] contains breaking changes
- [ ] adds new feature
- [x] modifies existing feature (bug fix or improvements)
- [ ] relies on other tasks
- [ ] documentation changes
- [ ] tested on mobile devices

**This pull request makes the following changes:**

**Changes**

- (ex: Change document B)

**To-dos**

- [x]  Basic layout and style update
- [x]  Reward animation
- [x] Use i18n
- [x]  Merge account unification
- [ ] Add to/remove from favourite button

ayumitk commented:
![268000190-077600b6-2007-4e2d-8eb5-93ae24b42941](https://github.com/AstarNetwork/astar-apps/assets/33358068/53be8a97-28f2-4b28-8415-2bee7c827186)

Kahonnohak commented:
Thanks Ayumi!

- Would it be possible to align all token amounts including ASTR? no preference of where but aligning with ASTR balance might make sense
- Could you also align the right side icons please?
- Also shall we use "Send" icon for ASTR transfer?
<img width="887" alt="Screenshot 2023-10-05 at 12 00 00" src="https://github.com/AstarNetwork/astar-apps/assets/77480847/46b46eff-707e-4189-b08f-1b28e992c45b">

- I have added a new colour #141E3E directly written sorry... later please add as Navy-4

- Some texts are default Black but please change them to Navy-1

- When it has the mobile header shall we use the mobile account style?
<img width="907" alt="Screenshot 2023-10-05 at 12 03 14" src="https://github.com/AstarNetwork/astar-apps/assets/77480847/7968da86-7915-4e8c-8729-45383590cd38">

- Now really want to change some icons BUT lets move on... I think after those changes we can invite Roy and Bobo to review this. There are funny behaviours at this moment but lets ask them to fix those. You could take a look at Reward animation.

Thanks

ayumitk commented:
@Kahonnohak 

> Would it be possible to align all token amounts including ASTR? no preference of where but aligning with ASTR balance might make sense

I aligned all token positions by the right side of symbols, but do you think style A or B would be better?

|Current|A|B|
|-|-|-|
|![assets-current](https://github.com/AstarNetwork/astar-apps/assets/33358068/3e6e552d-d1b0-4bdf-9a04-9b67b4d3ab7e)|![align-assets-a](https://github.com/AstarNetwork/astar-apps/assets/33358068/36d7daa6-9508-4f36-87b2-e037aafc4e56)|![align-assets-b](https://github.com/AstarNetwork/astar-apps/assets/33358068/c8660942-fd1e-43fa-b25f-bdf549195b77)|

ayumitk commented:
@Kahonnohak When the balance is 0, I need a faucet icon now. So I searched for faucet in [Flaticon](https://www.flaticon.com/search?word=faucet&color=black&shape=outline) (image attached), but I think these are too complicated. Do you have any idea about it?

<img width="870" alt="Screenshot 2023-10-05 at 12 52 55 PM" src="https://github.com/AstarNetwork/astar-apps/assets/33358068/d07d5d49-631c-454e-9070-180a295c60e0">

<img width="1049" alt="Screenshot 2023-10-05 at 12 57 28 PM" src="https://github.com/AstarNetwork/astar-apps/assets/33358068/5a04a33c-6404-495e-832b-91cf3448ba99">

Kahonnohak commented:
THANK YOU! Yes I meant A, is much easier to see numbers, although showing desimal are different each tokens... still A is good. Thank you so much.

And faucet..! forgotten about it. Let's go with this:
https://fonts.google.com/icons?selected=Material+Symbols+Rounded:water_drop:FILL@0;wght@300;GRAD@0;opsz@24&icon.query=water&icon.set=Material+Symbols&icon.style=Rounded
<img width="164" alt="Screenshot 2023-10-05 at 21 39 29" src="https://github.com/AstarNetwork/astar-apps/assets/77480847/28d71b4b-af48-4947-a2f9-a156e4aa6654">

ayumitk commented:
@Kahonnohak I adjusted all of the above and added an animation for the Reward circle. Please review it

Kahonnohak commented:
Nice work!! Lets review at creative sync later x

Kahonnohak commented:
Small favour for now sorry, but can we see background fixed at top version? what do you think?

impelcrypto commented:
@ayumitk 

Unify account
* Maybe implement it after merging [this](https://github.com/AstarNetwork/astar-apps/pull/925) PR? 

Add to/remove from favorite
* Maybe you store the data in localstorage like I did for `txHistories`. So the data object includes `address { network: { favoriteCoin } }`  Please check [this](https://github.com/AstarNetwork/astar-apps/blob/main/src/config/localStorage.ts#L25) file.

![image](https://github.com/AstarNetwork/astar-apps/assets/92044428/72907cd8-9403-4ca4-b249-ac441ba22b03)

ayumitk commented:
@impelcrypto Ah localstorage! thank you

Kahonnohak commented:
@ayumitk thanks a lot! could you quickly fix one thing - I cannot scroll down the mobile nav for this moment. could you take a look?

Kahonnohak commented:
@ayumitk all small things let me know if you don't have time

- Mobile header under line -> dark colour 1px
- Mobile evm network toggle -> move to the right
- Account Address -> font weight 700 and could you add network gradients if possible?
- Rewards numbers -> ASTR bottom like looks a bit below, small adjustment would be great
- Rewards animation -> A little slower - I think 16s look nice and peaceful
- Mobile padding in the asset list 32px
- Mobile assets search box -> please adjust - at least 16px apart?
- Dark colour asset list hover / locked token background colour -> 1F2F5F (Navy3)

<img width="388" alt="Screenshot 2023-11-15 at 13 01 52" src="https://github.com/AstarNetwork/astar-apps/assets/77480847/50b502d8-5443-429d-8062-fc47a90ef508">

impelcrypto commented:
@ayumitk 

> Could you add network change on this toggle please? Astar EVM(L1) <-> Astar zkEVM

You can add the button like:
Astar: disable the `Astar zkEVM` button for now.
Shibuya: just use <a> tag and go to 'zkatana-testnet/assets'
zKatana: just use <a> tag and go to 'shibuya-testnet/assets'

> Please also add claim function on the button on Native and Astar EVM.

You can refer to [this](https://github.com/AstarNetwork/astar-apps/blob/main/src/components/dapp-staking/my-staking/MyRewards.vue#L39) code

impelcrypto commented:
@ayumitk 
> You can add the button like:
Astar: disable the Astar zkEVM button for now.
Shibuya: just use a tag (link) and open 'zkatana-testnet/assets'
zKatana: just use a tag (link) and open 'shibuya-testnet/assets'

Maybe you can try like `providerEndpoints[endpointKey.ASTAR_ZKEVM].networkAlias` to get `zkatana-testnet/` value

Kahonnohak commented:
All accounts works like this - I hope it makes sense. I will add testnet and Shiden img later.
<img width="1032" alt="Screenshot 2023-11-16 at 15 26 16" src="https://github.com/AstarNetwork/astar-apps/assets/77480847/f032b16f-ca42-48f7-9bcb-6a89f954da0c">

ayumitk commented:
@Kahonnohak I've updated the following. Please review it 🙏

> I cannot scroll down the mobile nav for this moment. could you take a look?

> Could you add network change on this toggle please? Astar EVM(L1) <-> Astar zkEVM

> - Dark theme header under line -> 1px ?
> - Mobile evm network toggle -> move to the right
> - Account Address -> font weight 700 and could you add network gradients if possible?
> - Metamask icon -> squashed?
> - Rewards numbers -> ASTR line bottom looks not matching with the number, small adjustment would be great
> - Rewards animation -> A little slower - I think 16s look nice and peaceful
> - Mobile padding in the asset list 32px
> - Mobile assets search box -> please adjust - at least 16px apart?
> - Happy to hide the star feature for now
> - Dark colour asset list hover / locked token background colour -> 1F2F5F (Navy3)
> - Adding one more thing - wanted to change this top padding below the header. Seems 27px is a nice position.

Kahonnohak commented:
Thank you so much x

Kahonnohak commented:
@impelcrypto Please add network switch and claiming rewards logic

impelcrypto commented:
@ayumitk 
It looks cursor here isn't appropriate. The cursor is 'not-allowed' but I can click and claim the rewrds.
https://gyazo.com/bcbab020fdf42d7aa9ebbb43ca2a95b3

impelcrypto commented:
@ayumitk @Kahonnohak Is this PR still under 'Draft'?

ayumitk commented:
@Kahonnohak I've added the network switch part, but I still need to fix the rewards section logic.

@impelcrypto I've set this PR for ready. Let me ask about the rewards section logic at our sync later.

ayumitk commented:
@impelcrypto I created the rewards section by copying [this code](https://github.com/AstarNetwork/astar-apps/blob/main/src/components/dapp-staking/my-staking/MyRewards.vue). And I have 2 issues here.

1. The estimated reward amount will not show after refreshing the assets page. But once we go to the dapps staking page, the number will appear. So it seems we need to fetch/store/watch some data first on the assets page, but I don't know which one.
https://www.loom.com/share/f683109c077a475192e65791ea768f02?sid=a6d8810d-4260-48b5-a309-3bb4099f3e69

2. The disabling of the button condition will not disappear even if it's possible to claim. This might related to the first issue.

The disabling of the button condition will not disappear even if it's possible to claim.

impelcrypto commented:
@ayumitk 
I can see the faucet button instead of transfer button here. You might want to check the condition for Shibuya. 
![image](https://github.com/AstarNetwork/astar-apps/assets/92044428/7cf8b3f4-0588-4b2d-9a7c-8fd5eea3cd78)

Kahonnohak commented:
Last request before going production please!

@impelcrypto please if @ayumitk did not have time
1) When a user do not have staking, please say `Start staking` and navigate to dApp Staking page.
2) When user has staking but no rewards to be claimed, please show 0 ASTR - at this moment, portal seems to be trying to find number.
3) Mobile staking box is not fit in the width. please change.
4) Please change the network toggle font to 12px
5) Please add account background for testnets (Shibuya and zKatana) also Shiden. (accound_bg_testnet/shiden) I added to img folder already.

I have checked most browsers and android/iphone. All look great!!
<img width="468" alt="Screenshot 2023-11-21 at 13 56 06" src="https://github.com/AstarNetwork/astar-apps/assets/77480847/60c1d4db-bf5f-4a40-a34a-cea2ece6a4e1">
<img width="665" alt="Screenshot 2023-11-21 at 13 54 09" src="https://github.com/AstarNetwork/astar-apps/assets/77480847/ae0da96f-c794-491b-a70d-464399ce14e7">

ayumitk commented:
> I can see the faucet button instead of transfer button here. You might want to check the condition for Shibuya.

@impelcrypto It seems there is the same bug on the production. Could you please fix this?

<img width="1712" alt="Screenshot 2023-11-22 at 12 52 22 AM" src="https://github.com/AstarNetwork/astar-apps/assets/33358068/1e9728d5-b6dd-4f9b-9454-a1a509c0c9ba">

ayumitk commented:
@impelcrypto could you take care of 1 and 2 please?

> 1. When a user do not have staking, please say Start staking and navigate to dApp Staking page.
> 2. When user has staking but no rewards to be claimed, please show 0 ASTR - at this moment, portal seems to be trying to find number.

@Kahonnohak I've fixed 3, 4 and 5!

> 3. Mobile staking box is not fit in the width. please change.
> 4. Please change the network toggle font to 12px
> 5. Please add account background for testnets (Shibuya and zKatana) also Shiden. (accound_bg_testnet/shiden) I added to img folder already.

impelcrypto commented:
> It seems there is the same bug on the production. Could you please fix this?

Production is Ok because the transfer button shows after expanding it. How about adding the transfer button next to the faucet button if the portal connects Shibuya?

<img width="1162" alt="image" src="https://github.com/AstarNetwork/astar-apps/assets/92044428/031046e1-a52b-4b1d-88a4-b43bb588d122">
.

Kahonnohak commented:
Thank you Ayumi!

Yes, it is only Shibuya that has faucet and transfer at the same time, and anyways this should be the place for all action buttons:)
<img width="959" alt="Screenshot 2023-11-21 at 21 27 45" src="https://github.com/AstarNetwork/astar-apps/assets/77480847/dd0ee394-eb1a-4074-899f-35504be76371">

ayumitk commented:
@impelcrypto @Kahonnohak I've added a transfer button for shibuya in [this](https://github.com/AstarNetwork/astar-apps/pull/956/commits/57d2f8421786e2291a169f98930377792e415579) commit.

impelcrypto commented:
@ayumitk 
> @impelcrypto @Kahonnohak I've added a transfer button for shibuya in [this](https://github.com/AstarNetwork/astar-apps/pull/956/commits/57d2f8421786e2291a169f98930377792e415579) commit.

Looks good. I think the 'expend' icon is missing for mobile devices.
https://gyazo.com/3a46a2124127c3e2017bc993a88595e8

Kahonnohak commented:
> @ayumitk
> 
> > @impelcrypto @Kahonnohak I've added a transfer button for shibuya in [this](https://github.com/AstarNetwork/astar-apps/pull/956/commits/57d2f8421786e2291a169f98930377792e415579) commit.
> 
> Looks good. I think the 'expend' icon is missing for mobile devices, or is this according to the figma design? https://gyazo.com/3a46a2124127c3e2017bc993a88595e8

Oh sorry I missed this and it looks like the locked balance is missing the vesting tokens

<img width="834" alt="Screenshot 2023-11-22 at 09 33 50" src="https://github.com/AstarNetwork/astar-apps/assets/77480847/27784d7c-e163-43c7-802c-0112d43104f0">

Kahonnohak commented:
I see something went wrong. I'll check everything now
<img width="423" alt="Screenshot 2023-11-22 at 09 40 44" src="https://github.com/AstarNetwork/astar-apps/assets/77480847/9c6805b5-5ab6-430f-8f0a-0b6a2bfc16fc">

Kahonnohak reviewed (approved):
LGTM!! Thank you so much for the great work!

bobo-k2 commented on src/components/assets/Account.vue:
```diff
@@ -240,6 +273,27 @@ export default defineComponent({
+    };
+
+    const bg = computed<String>(() => {
+      if (currentNetworkIdx.value === 0) {
```
Use `endpointKey` enum instead numbers

ayumitk replied:
Fixed it in this [commit](https://github.com/AstarNetwork/astar-apps/pull/956/commits/0c5ca2f0a0b5e792786a55fe339f2ae21fcdbbe4)

bobo-k2 commented on src/components/assets/Assets.vue:
```diff
@@ -167,6 +156,20 @@ export default defineComponent({
       window.removeEventListener(event, handler);
     });
 
+    const isDarkTheme = computed<boolean>(() => store.getters['general/theme'] === 'DARK');
```
We have this line in so many places in our codebase. What do you think moving the line to some hook? It doesn't need to be in this PR

ayumitk replied:
Let me work on this in another PR. I'll do it next week

bobo-k2 commented on src/components/assets/Erc20Currency.vue:
```diff
@@ -1,118 +1,145 @@
+            </q-tooltip>
+          </div>
+
+          <!-- <div>
```
Delete if not needed

bobo-k2 commented on src/components/assets/EvmAssetList.vue:
```diff
@@ -97,6 +106,13 @@ export default defineComponent({
     };
 
+    const store = useStore();
+    const isDarkTheme = computed<boolean>(() => store.getters['general/theme'] === 'DARK');
```
Already commented above

bobo-k2 commented on src/components/assets/EvmNativeToken.vue:
```diff
@@ -1,69 +1,115 @@
-              class="btn btn--sm"
-              @click="handleModalFaucet({ isOpen: true })"
+              <path
+                d="M11.9 17.61C11.18 17.66 10.36 17.44 9.47999 16.95C8.62999 16.47 8.07999 15.61 7.86999 14.4C7.81999 14.16 7.71999 13.97 7.55999 13.83C7.19999 13.51 6.63999 13.54 6.31999 13.91C6.13999 14.12 6.07999 14.37 6.13999 14.68C6.45999 16.39 7.23999 17.63 8.45999 18.37C9.55999 19.04 10.64 19.37 11.69 19.37C11.79 19.37 11.89 19.37 11.99 19.37C12.24 19.35 12.44 19.25 12.6 19.09C12.77 18.92  ...
```
Does it make sense to create svg file instead hadrdcoding path in the component? If we move svg to a separate file we have a reusable component

ayumitk replied:
I've moved svg icon to Astar UI and replaced it in this [commit](https://github.com/AstarNetwork/astar-apps/pull/956/commits/ee35c19b4b8422c31510093c4f2735c7b75bf87e)

bobo-k2 commented on src/components/header/TroubleHelp.vue:
```diff
@@ -1,16 +1,24 @@
+          xmlns="http://www.w3.org/2000/svg"
+        >
+          <path
+            d="M14.6048 7.37424C14.6048 6.60107 14.3463 5.98007 13.8292 5.51124C13.3121 5.04242 12.6296 4.80801 11.7817 4.80801C11.2448 4.80801 10.7689 4.91756 10.3539 5.13666C9.93891 5.35577 9.58069 5.68563 9.27922 6.12624C9.10147 6.37715 8.86397 6.52541 8.56672 6.57103C8.26946 6.61665 8.00974 6.54402 7.78755 6.35315C7.623 6.21028 7.52901 6.03109 7.50559 5.8156C7.48217 5.60011 7.53232 5.39573 7. ...
```
svg path, see the comment above

ayumitk replied:
I've moved svg icon to Astar UI and replaced it in this [commit](https://github.com/AstarNetwork/astar-apps/pull/956/commits/ee35c19b4b8422c31510093c4f2735c7b75bf87e)

impelcrypto reviewed (approved):
LGTM
@ayumitk Let's move some icons to Astar-UI later

ayumitk commented on src/components/assets/Account.vue:
```diff
@@ -1,87 +1,118 @@
+                <astar-icon-person />
+              </button>
+              <q-tooltip>
+                <span class="text--tooltip">Unify accounts</span>
```
Fixed it in this [commit](https://github.com/AstarNetwork/astar-apps/pull/956/commits/0b71f3a800b60d1bae2a14f328a21aaea62291e1)

ayumitk commented on src/components/assets/Account.vue:
```diff
@@ -1,87 +1,118 @@
-                </q-tooltip>
+              <div class="text--balance">
+                {{ $n(totalBal) }}
+                <span>USD</span>
```
Fixed it in this [commit](https://github.com/AstarNetwork/astar-apps/pull/956/commits/0b71f3a800b60d1bae2a14f328a21aaea62291e1)

ayumitk commented on src/components/assets/Account.vue:
```diff
@@ -1,87 +1,118 @@
+            <!-- Native -->
+            <div v-else class="btn active">
+              {{ currentNetworkIdx === 4 ? 'Astar' : currentNetworkName.replace('Network', '') }}
+              Native
```
Fixed it in this [commit](https://github.com/AstarNetwork/astar-apps/pull/956/commits/0b71f3a800b60d1bae2a14f328a21aaea62291e1)

ayumitk commented on src/components/assets/Account.vue:
```diff
@@ -1,87 +1,118 @@
+                  EVM (L1)
+                </div>
+                <a v-if="currentNetworkIdx === 2" class="btn" href="/zkatana-testnet/assets">
+                  Astar zkatana
```
Fixed it in this [commit](https://github.com/AstarNetwork/astar-apps/pull/956/commits/0b71f3a800b60d1bae2a14f328a21aaea62291e1)

ayumitk commented on src/components/assets/Account.vue:
```diff
@@ -1,87 +1,118 @@
+              <!-- zkEVM -->
+              <template v-if="isZkEvm">
+                <a class="btn" href="/shibuya-testnet/assets"> Shibuya EVM (L1) </a>
+                <div v-if="isZkEvm" class="btn active">Astar zkatana</div>
```
Fixed it in this [commit](https://github.com/AstarNetwork/astar-apps/pull/956/commits/0b71f3a800b60d1bae2a14f328a21aaea62291e1)

ayumitk commented:
> Let's move some icons to Astar-UI later.

@impelcrypto I've moved svg icon to Astar UI and replaced them in this [commit](https://github.com/AstarNetwork/astar-apps/pull/956/commits/ee35c19b4b8422c31510093c4f2735c7b75bf87e)

bobo-k2 reviewed (approved):
LGTM
